### PR TITLE
Add support for speed check against ufs with hidden credentials

### PR DESCRIPTION
**Pull request by yuzhu**
Speed check can not be performed on a UFS right now because ufs credentials are masked when they are returned from the master.
This works around that issue by calling into the file master to launch a speed test which will run as a job.

### CODE REVIEW

File: job/server/pom.xml [hunk 1]
```diff
@@ -87,7 +87,7 @@
     </dependency>
     <dependency>
       <groupId>org.alluxio</groupId>
-      <artifactId>alluxio-stress-common</artifactId>
+      <artifactId>alluxio-stress-shell</artifactId>
```
Comment (madanadit) [inline job/server/pom.xml:90]: why is this needed?

File: core/server/master/pom.xml [hunk 2]
```diff
@@ -97,6 +97,11 @@
       <artifactId>alluxio-job-client</artifactId>
       <version>${project.version}</version>
     </dependency>
+    <dependency>
+      <groupId>org.alluxio</groupId>
+      <artifactId>alluxio-stress-shell</artifactId>
```
Comment (madanadit) [inline core/server/master/pom.xml:102]: @yuzhu the flow as of now is the new master RPC depend on the shell module to execute commands which end up making RPCs to the job master process?

**Review by madanadit — COMMENTED** [review]
Thanks @yuzhu. Like I mentioned before, the pom dependency feels weird but looks good otherwise. Requested review from Gene and Jiacheng to see if there is a better way.

File: core/server/master/pom.xml (hunk 2 above)
Comment (yuzhu) [reply core/server/master/pom.xml:102]: yes

Comment (jiacheliu3) [reply core/server/master/pom.xml:102]: Yea currently this is the only way.

**Review by jiacheliu3 — APPROVED** [review]
LGTM except the questions on job server pom from Adit. Thanks for the improvement!

File: job/server/pom.xml (hunk 1 above)
Comment (yuzhu) [reply job/server/pom.xml:90]: reverted

**Review by madanadit — APPROVED** [review]

**Review by gpang — COMMENTED** [review]
@yuzhu I don't like that we have to add an RPC endpoint to FSM to just run this specific test. It feels too specific and too much like a hack. Instead, can we figure out how we can pass mount configuration to a stress job?

For example, is there something we can do in `StressBenchDefinition` to get and pass along mount config?

**Comment by yuzhu** [comment]
@gpang I think the central issue here is that `UFSIOBench` directly operates on the `UnderFileSystem` class, and that's why it needs the credentials. I suppose we can read and write through the master, but that will surely involve other components of the alluxio master such as metadata ops and is not a pure UFS IO benchmark.

**Comment by gpang** [comment]
I think JobWorkers do have access to the mount points, via JobServerContext and the UfsManager (https://github.com/Alluxio/alluxio/blob/d8193189a3de15cc01cfb315cb9c981b08b13e3e/job/server/src/main/java/alluxio/job/JobServerContext.java#L31)

We just need to figure out how that information can be propagated to the task itself.

**Comment by yuzhu** [comment]
@gpang The task can get an instance of the `UfsManager`, but then it needs to do a shell exec to launch `UfsIOBench`, so this still wouldn't work. We need to somehow allow `UfsManager` to return credentials to make this work. so we can pass it to the shell execute.

**Comment by yuzhu** [comment]
See #12480 for a version without POM changes and going through the FileSystemMaster
